hooks/cache_location.py

```diff
@@ -126,23 +114,6 @@ def _ensure_file_exists(self, path):
                     raise TankError("Could not create cache file '%s': %s" % (path, e))
             finally:
                 os.umask(old_umask)
-    
+
     def _ensure_folder_exists(self, path):
-        """
-        Helper method - creates a folder if it doesn't already exists
-        
-        :param path: path to create
-        """
-        if not os.path.exists(path):
-            old_umask = os.umask(0)
-            try:
-                os.makedirs(path, 0777)
-            except OSError, e:
-                # Race conditions are perfectly possible on some network storage setups
-                # so make sure that we ignore any file already exists errors, as they 
-                # are not really errors!
-                if e.errno != errno.EEXIST: 
-                    raise TankError("Could not create cache folder '%s': %s" % (path, e))
-            finally:
-                os.umask(old_umask)
-            
+        sgtk.util.path.ensure_path_exists(path)
```

Comment:
So I think this folder is a bit special and I am not totally sure we should have a special method for it - but happy to discuss! The new `sgtk.util.path.ensure_path_exists()` defaults to `0777` which maybe is too open for this stuff? Should the stuff we are caching be accessible only by the current user by default (e.g. `0700`) ? 

The previous idea with for example the `bundle.ensure_folder_exists()` method was that apps could create folders on disk and that the behaviour for permissions etc is controlled via a core hook, allowing an admin to control permissions quickly on a global level!

This feels a little different and I wonder if a better option would be to simply make sure that `tk.local_site_cache_location` always returns a path that is guaranteed to exist on disk. And the two folders added explicitly in the hook are also created inside the hook (the project_xxx, config_xxx) - using explicit I/O operations.

This means that toolkit is responsible (both generating and creating the path) for the portion that it dictates (`tk.local_site_cache_location`) and the hook is responsible for the part that runs in the hook.

How does that sound?